src/android/Geolocation.java

```diff
@@ -190,7 +197,12 @@ private void addWatch(LocationContext locationContext) {
 
     @SuppressLint("MissingPermission")
     private void requestLocationUpdates(LocationContext locationContext, LocationRequest request) {
-        fusedLocationClient.requestLocationUpdates(request, locationContext.getLocationCallback(), null);
+        if (isConnectedToWifiOrData(cordova.getContext())){
+            fusedLocationClient.requestLocationUpdates(request, locationContext.getLocationCallback(), null);
+        }
+        else {
+            locationHelper.requestGPSLocationUpdates(0, 0, cordova.getContext(), locationContext.getLocationListener());
```

Comment:
Can we leave a comment here explaining what are these 0 values, as they aren't self-explanatory?

Reply:
That's a good point. I left comments for both values.